lib/build.js

```diff
@@ -517,6 +527,36 @@ class BuildModel extends BaseModel {
         return ['ABORTED', 'FAILURE', 'SUCCESS'].includes(this.status) ||
         (this.status === 'UNSTABLE' && !!this.endTime);
     }
+
+    /**
+     * getMetrics for this build
+     * @method getMetrics
+     * @param  {Object}   [config]              Configuration object
+     * @param  {String}   [config.startTime]    Look at steps created after this startTime
+     * @param  {String}   [config.endTime]      Look at steps created before this endTime
+     * @return {Promise}  Resolves to array of metrics for steps belong to this event
+     */
+    async getMetrics(config) {
+        // if no config pass in then get all (will not pass startTime/endTime to datastore)
+        const startTime = config ? config.startTime : null;
+        const endTime = config ? config.endTime : null;
+
+        // Get steps during this time range
+        const steps = await this.getSteps({
+            startTime,
+            endTime
+        });
+
+        // Generate metrics
+        const metrics = steps.map((s) => {
+            const { id, name, code } = s;
```

Comment:
```suggestion
            const { id, name, code, endTime, startTime } = s;
            const duration = Math.round((new Date(endTime) - new Date(startTime)) / 1000);
```
not sure if this is any better

Reply:
ah couldn't do that before because it will conflict with the `startTime` and `endTime` outside. But can do it now 👍